extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/configuration_validation/ClassWithConfigProperties.java

```diff
@@ -0,0 +1,69 @@
+package io.quarkus.hibernate.validator.runtime.configuration_validation;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.enterprise.inject.spi.CDI;
+import javax.validation.ConstraintViolation;
+import javax.validation.ConstraintViolationException;
+import javax.validation.Validator;
+
+/**
+ * Holds the class and the configuration properties i.e class fields annotated with
+ * {@link org.eclipse.microprofile.config.inject.ConfigProperty}
+ */
+public class ClassWithConfigProperties {
+    private final Class<?> aClass;
+    private final List<String> configurationProperties;
+    static List<ClassWithConfigProperties> classWithConfigurationProperties;
+
+    ClassWithConfigProperties(Class<?> aClass, List<String> configurationProperties) {
+        this.aClass = aClass;
+        this.configurationProperties = configurationProperties;
+    }
+
+    /**
+     * Throw an {@link ConstraintViolationException} when a configuration property is not valid
+     */
+    void validate(Validator validator) {
+        for (Object obj : CDI.current().select(aClass)) {
+            for (String property : configurationProperties) {
```

Comment:
Is there a reason to validate each property on it's own and not just validate the whole class?

Reply:
Oh yes, we would not want to throw an exception for the wrong reasons i.e class might contain other  fields (not config property) that needs to be validated. 

That said, I was undecided between validating at once and construct the message only considering constraints violation for config properties, or this current behaviour. I kept he behaviour for clarity also, i.e `only validate a config properties`

Reply:
@gsmet is the expert here, so let's see what he says :)